app/src/pages/eligibility.tsx

```diff
@@ -9,73 +8,104 @@ import ButtonLink from '@components/ButtonLink'
 import InputChoiceGroup from '@components/InputChoiceGroup'
 import RequiredQuestionStatement from '@components/RequiredQuestionStatement'
 
-interface Props {
-  previousRoute: string
-}
-
-const Eligibility: NextPage<Props> = (props: Props) => {
-  const incomeRoute = '/income'
-  const { session, setSession } = useAppContext()
-  const [continueBtn, setContinueBtn] = useState({
-    labelKey: 'continue',
-    route: incomeRoute,
-  })
-  const [form, setForm] = useState(session?.eligibility)
-  const requiredMet = () => {
-    const categorical = [
-      'pregnant',
-      'baby',
-      'child',
-      'guardian',
-      'pregnant',
-      'none',
-    ].some((category) => form[category as keyof typeof form])
-    const programs = ['insurance', 'snap', 'tanf', 'none2'].some(
-      (program) => form[program as keyof typeof form]
-    )
-
-    return form.residential && categorical && form.before && programs
-  }
-  const [disabled, setDisabled] = useState<boolean>(!requiredMet())
+import type { EditablePage, EligibilityData } from '@src/types'
+import {
+  isValidEligibility,
+  validEligibilityOptions,
+} from '@utils/dataValidation'
+import { initialEligibilityData } from '@utils/sessionData'
 
+const Eligibility: NextPage<EditablePage> = (props: EditablePage) => {
+  // Get the session from props.
+  const {
+    session,
+    setSession,
+    reviewMode = false,
+    backRoute = '',
+    forwardRoute = '',
+  } = props
+  // Initialize form as a state using blank values.
+  const [form, setForm] = useState<EligibilityData>(initialEligibilityData)
+  // Use useEffect() to properly load the data from session storage during react hydration.
   useEffect(() => {
-    /* NOTE: We are using useEffect() because we want to make sure the props provided by getServerSideProps() are reliably loaded into the page. */
-    const prevRouteIndex = props.previousRoute.lastIndexOf('/')
-    const previousRoute = props.previousRoute.substring(prevRouteIndex)
-    if (form.none) {
-      setContinueBtn({ ...continueBtn, route: '/other-benefits' })
-    } else if (previousRoute === '/review') {
-      setContinueBtn({
-        labelKey: 'updateAndReturn',
-        route: previousRoute,
-      })
-    } else setContinueBtn({ ...continueBtn, route: incomeRoute })
-  }, [form.none, props.previousRoute])
+    setForm(session.eligibility)
+  }, [session.eligibility])
 
+  // Set up action button and routing.
+  const actionButtonLabel = reviewMode ? 'updateAndReturn' : 'continue'
+
+  // Set a state for whether the form requirements have been met and the
+  // form can be submitted. Otherwise, disable the submit button.
+  const [disabled, setDisabled] = useState(true)
+  // Use useEffect() to properly load the data from session storage during react hydration
```

Comment:
We also have this explanation comment on line 29 right? I know we're commenting on a different effect but at least for me I wasn't sure I got value from reading the same comment twice 😬 hopefully not too nitty.

I haven't looked at the full diff in this PR yet but my initial reaction is that using local storage to persist state for the user across refresh is an okay pattern that I've seen before (alternatives could be pulling from the backend or query params probably tradeoffs to each). 

In terms of best practices something I'm wondering on initial read is if pulling out some of the form logic into a custom hook like `useFormState` could be helpful - then the hook could return values for form as well as disabled, change handler etc which might be really nice to have a home to hold some of the comments explaining how things are working etc. 

Would be particularly helpful if it's not just eligibility page that will be using the form state? I think there is an example of what this could look like from PFML, although it doesn't have the local storage piece (and is older code style using this etc)

https://github.com/navapbc/archive-massgov-pfml/blob/aligg/pfml/portal/src/hooks/useFormState.tsx

Reply:
Excellent feedback. Thank you! Not too nitty at all.

I'll also think about the idea of pulling out the form logic into a custom hook.